### PR TITLE
Update sockets to support IPV6

### PR DESCRIPTION
**Pull request by AdrianSoundy**
## Description
This change goes with the interpreter update for IDF 5.1.1
It updates the socket code to support working with IPV6 for support of OpenThread and for general use.

## How Has This Been Tested?
- Tested changes in IPAddress to support IPV6 format addresses for Parsing and Outputting addresses.
- Tested with some local UDP and TCPIP socket code

TODO 
- Add some test cases for IPV6

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:
- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by AdrianSoundy** [comment]
@josesimoes 
Got a weird SonarCloud error.  Not sure how to fix that.
I says need to add validation on constructor but its already calling basic constructor with the validation.

**Comment by AdrianSoundy** [comment]
Looked at adding some IPV6 tests but couldn't get existing ones to run on device. Never seems to deploy.
Followed reference here:
https://docs.nanoframework.net/content/unit-test/running-tests.html#:~:text=Running%20the%20tests%20on%20a%20real%20hardware%20You%27ll,deployed%20into%20the%20device%20attached%20to%20your%20machine.

I see other PR for this assemblies has some changes to tests. I'll wait until that's been merged.

**Review by josesimoes — APPROVED** [review]
LGTM!

**Pull request merged**
